Api/V8/JsonApi/Helper/AttributeObjectHelper.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Api\V8\JsonApi\Helper;
+
+use Api\V8\BeanDecorator\BeanManager;
+use Api\V8\JsonApi\Response\AttributeResponse;
+
+class AttributeObjectHelper
+{
+    /**
+     * @var BeanManager
+     */
+    private $beanManager;
+
+    /**
+     * @param BeanManager $beanManager
+     */
+    public function __construct(BeanManager $beanManager)
+    {
+        $this->beanManager = $beanManager;
+    }
+
+    /**
+     * @param \SugarBean $bean
+     * @param array|null $fields
+     *
+     * @return AttributeResponse
+     */
+    public function getAttributes(\SugarBean $bean, $fields = null)
+    {
+        $bean->fixUpFormatting();
+        $attributes = $bean->toArray();
+
+        // using the ISO 8601 format for dates
+        array_walk($attributes, function (&$value) {
```

Comment:
array_walk returns could be TRUE on success or FALSE on failure. Your code does not handle the failure.

Reply:
Yeah, that's true, but not because of the returns value. I both want to iterate and change the value if it's necessary and array_walk could do the job as well, but you are right.  Gonna use array_map instead,

Reply:
Fixed.